Add Play-framework support

woutersterp opened the pull request:
Add play-framework support to Tsuru

cezarsa commented:
Thats's really great! Could you also create a small example application and put it in `examples/play`? It should simply print `"Hello world from tsuru"` when called.

woutersterp commented:
I've added a play-framework example app.

cezarsa commented on play/install:
```diff
@@ -0,0 +1,19 @@
+
+/var/lib/tsuru/base/install
+
+apt-get install software-properties-common -y
```
You'll need to add `unzip` here. Testing this I found that it's not available by default.

cezarsa commented:
I got the following error trying to run the test application:
```
...
remote: /home/application/current/build.sbt:5: error: not found: value PlayScala
remote: lazy val root = (project in file(".")).enablePlugins(PlayScala)
remote:                                                      ^
remote: sbt.compiler.EvalException: Type error in expression
remote: 	at sbt.compiler.Eval.checkError(Eval.scala:343)
remote: 	at sbt.compiler.Eval.compileAndLoad(Eval.scala:165)
remote: 	at sbt.compiler.Eval.evalCommon(Eval.scala:135)
remote: 	at sbt.compiler.Eval.evalDefinitions(Eval.scala:109)
remote: 	at sbt.EvaluateConfigurations$.evaluateDefinitions(EvaluateConfigurations.scala:197)
...
remote: [error] sbt.compiler.EvalException: Type error in expression
```

Full log is here: https://gist.github.com/cezarsa/15a5d3ac422f30dbdb27

Any ideas about what might be wrong?

woutersterp commented:
I fixed the plugin problem, also I updated the java-version from openjdk-7 to openjdk-8 and apparently the're is a chmod needed on the scripts to work, so I added that too. 

SBT is a bit slow in resolving the dependencies sometimes, so be patient when deploying the example app. As long SBT it's spamming messages to the console it's working. 

There's still a minor issue with permissions which I will try to fix tomorrow.

woutersterp commented:
It now is working fine from my repository, so it should be working fine on yours too, would you test it again?

Sorry for the inconvenience, I had tested it with a zip-download from Dropbox. I never thought a download from Github was any different and made some wrong assumptions.  

If you've any issues you can reply to this message. I will try to fix them as soon as possible.

cezarsa commented:
No problem at all, it's great your working on this platform. I'm building this locally and I'll merge as soon as it's running.

cezarsa commented on play/Dockerfile:
```diff
@@ -0,0 +1,14 @@
+run tar -xvf basebuilder.tar.gz -C /var/lib/tsuru --strip 1
+run cp /var/lib/tsuru/play/deploy /var/lib/tsuru
+run cp /var/lib/tsuru/base/start /var/lib/tsuru
+run chmod -R 777 /var/lib/tsuru
```
I don't think this is necessary, maybe you got into problems because play/deploy and play/install files were committed without +x. You can solve this by running `chmod +x deploy install` locally and then committing and pushing these files.

woutersterp replied:
Hi @cezarsa I'm afraid it's not possible to chmod the files and commit them, since I'm on a Windows-machine. Also chmod +x /var/lib/tsuru works for installing, but on deploy I got errors copying the Procfile from /home/application/current, since it's nice-to-have a editable Procfile on a Play-project for additional (alternative-)configuration.

You may try to chmod the scripts, but as stated above, I'm not sure it resolves all the issues.

Thanks in advance for your reply!

woutersterp replied:
I've chmodded the files on a Ubuntu VM. I'll check if it works now without the chmod in code.

woutersterp commented:
I don't know what is wrong with the chmod, but it doesn't seem to work. The only way it works is by using the chmod 777.

cezarsa commented on play/deploy:
```diff
@@ -0,0 +1,13 @@
+echo "Procfile:"
+cat /home/application/current/Procfile
+
+cp /home/application/current/Procfile ${SOURCE_DIR}/utils/Procfile
```
This line has to be removed. There's no need for it, tsuru will already use the Procfile sent by the application.
Without it everything worked fine.

cezarsa commented:
Yay it's working. I just had to remove the line I commented upon.

woutersterp commented:
Nice! Should I change my fork? Or do you commit your changes to the master?

cezarsa commented:
I can do this after merging. Thanks!
